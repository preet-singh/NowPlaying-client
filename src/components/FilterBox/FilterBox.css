

@media screen and (min-width: 320px) and (max-width: 768px){
  #filterForm {
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin: 1em;
    margin-left: auto;
    margin-right: auto;
    width: 75%;
    border: 2px outset white;
    padding: 1em;

  }

  #filter-buttons{
    margin: 0.8em;
  }

  #filterForm button {
    width: 35%;
  }
}

@media screen and (min-width: 768px) {
  #filterForm {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: 1em;
    margin-left: auto;
    margin-right: auto;
    width: 55%;
    border: 2px outset white;
    border-radius: 3px;
    padding: 1em;
  }

  #filterForm label {
    margin: 0.5em;
  }

  #filterForm button {
    width: 100px;
    height: 3em;
    margin-left: 0.5em;
  }
}