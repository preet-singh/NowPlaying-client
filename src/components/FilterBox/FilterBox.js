//Dependencies
import React, { Component } from 'react';
import UserContext from '../../utils/context'

class FilterBox extends Component {
  static contextType = UserContext;

  constructor(props){
    super(props)
    this.state = {
      minimumYear: '',
      maximumYear: '',
      genre: '',
      rating: '',
      familyRating: '',
      error: null,
      originalCategoryItems: []
    }
  }
  
  componentDidMount() {
    this.setState({
      originalCategoryItems: this.context.filteredCategoryItems
    })
  }
  
  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = (e, currentCategoryItems) => {
    e.preventDefault();
    this.context.setFilteredCategoryItems(currentCategoryItems);
    this.setState({
      minimumYear: '',
      maximumYear: '',
      genre: '',
      rating: '',
      familyRating: ''
    })
  }

  handleClear = e => {
    e.preventDefault();
    this.setState({
      minimumYear: '',
      maximumYear: '',
      genre: '',
      rating: '',
      familyRating: '',
      originalCategoryItems: this.state.originalCategoryItems
    })
    this.context.setFilteredCategoryItems(this.context.categoryItems);
    this.context.setSearchedCategoryItems(this.context.categoryItems);
  }

  render() {
    let currentCategoryItems = this.state.originalCategoryItems;

    if(this.state.minimumYear){
      currentCategoryItems = currentCategoryItems.filter(item => Number(this.state.minimumYear) <= Number(item.release_date.slice(0,4)))
    }
    if(this.state.maximumYear){
      currentCategoryItems = currentCategoryItems.filter(item => Number(this.state.maximumYear) >= Number(item.release_date.slice(0,4)))
    }
    if(this.state.genre) {
      currentCategoryItems = currentCategoryItems.filter(item => item.genre === this.state.genre);
    }
    if(this.state.rating) {
      currentCategoryItems = currentCategoryItems.filter(item => Number(item.imdb_rating) >= Number(this.state.rating));
    }
    if(this.state.familyRating) {
      currentCategoryItems = currentCategoryItems.filter(item => item.mpaa_rating === this.state.familyRating);
    }
    
    return (
      <form id="filterForm" name="filterForm" onSubmit={e => {this.handleSubmit(e,currentCategoryItems)}}>
        <label htmlFor="minimumYear" id="minimumYearLabel" name="minimumYearLabel">Earliest Year:</label>
        <input 
          type="text" 
          placeholder="1920" 
          id="minimumYear" 
          name="minimumYear" 
          onChange={this.handleChange} 
        />
        <label htmlFor="latestYear" id="latestYearLabel" name="latestYearLabel">Latest Year:</label>
        <input 
          type="text" 
          placeholder={new Date().getFullYear()}
          id="latestYear"
          name="maximumYear" 
          onChange={this.handleChange} 
        />
        <label htmlFor="genre" id="genreLabel" name="genreLabel">Genre:</label>
        <input 
          type="text" 
          placeholder="Romance"
          name="genre"
          onChange={this.handleChange}   
        />
        <label htmlFor="rating" id="ratingLabel" name="ratingLabel">Minimum Rating:</label>
        <select 
          id="rating" 
          name="rating" 
          defaultValue="1"
          onChange={this.handleChange} 
        >
          <option value="10">10</option>
          <option value="9">9</option>
          <option value="8">8</option>
          <option value="7">7</option>
          <option value="6">6</option>
          <option value="5">5</option>
          <option value="4">4</option>
          <option value="3">3</option>
          <option value="2">2</option>
          <option value="1">1</option>
        </select>
        <label htmlFor="rating" id="ratingLabel" name="ratingLabel">Highest Family Rating:</label>
        <select 
          id="familyRating" 
          name="familyRating" 
          defaultValue="R"
          onChange={this.handleChange} 
        >
          <option value="G">G</option>
          <option value="PG">PG</option>
          <option value="PG-13">PG-13</option>
          <option value="R">R</option>
        </select>
        <button type="submit">Submit</button>
        <button type="button" onClick={(e) => this.handleClear(e)}>Clear</button>
      </form>
    );
  }
}
export default FilterBox;