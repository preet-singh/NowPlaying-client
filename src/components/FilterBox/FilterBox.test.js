import React from 'react';
import ReactDOM from 'react-dom';
import FilterBox from './FilterBox';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Filter box testing', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <MemoryRouter>
            <FilterBox />
        </MemoryRouter>, 
    div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders the filter box UI', () => {
    const wrapper = shallow(<FilterBox />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
});