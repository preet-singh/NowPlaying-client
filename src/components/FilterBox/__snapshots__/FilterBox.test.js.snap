// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter box testing renders the filter box UI 1`] = `
<form
  id="filterForm"
  name="filterForm"
  onSubmit={[Function]}
>
  <label
    htmlFor="minimumYear"
    id="minimumYearLabel"
    name="minimumYearLabel"
  >
    Earliest Year:
  </label>
  <input
    id="minimumYear"
    name="minimumYear"
    onChange={[Function]}
    placeholder="1920"
    type="text"
  />
  <label
    htmlFor="latestYear"
    id="latestYearLabel"
    name="latestYearLabel"
  >
    Latest Year:
  </label>
  <input
    id="latestYear"
    name="maximumYear"
    onChange={[Function]}
    placeholder={2019}
    type="text"
  />
  <label
    htmlFor="rating"
    id="ratingLabel"
    name="ratingLabel"
  >
    Minimum Rating:
  </label>
  <select
    defaultValue="1"
    id="rating"
    name="rating"
    onChange={[Function]}
  >
    <option
      value="10"
    >
      10
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="1"
    >
      1
    </option>
  </select>
  <div
    id="filter-buttons"
  >
    <button
      className="black-button margin10"
      type="submit"
    >
      Submit
    </button>
    <button
      className="black-button margin10"
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </div>
</form>
`;
